addons/onboarding/models/onboarding_step.py

```diff
@@ -61,7 +62,7 @@ def action_set_just_done(self):
         # Make sure progress records exist for the current context (company)
         steps_without_progress = self.filtered(lambda step: not step.current_progress_step_id)
         steps_without_progress._create_progress_steps()
-        return self.current_progress_step_id.action_set_just_done()
+        return self.current_progress_step_id.action_set_just_done().step_id
```

Comment:
Isn't it an equivalent of returning self? This confuses me ^^
Or maybe not as we only return the ones for which it succeeded?

Looks like all we want to know is if this marked the step(s) as done so maybe this is easier to understand?

```py
marked_as_done = self.current_progress_step_id.action_set_just_done()
return self == marked_as_done.step_id  # return True if all steps in self were marked as done
```

Reply:
Since "action_set_just_done" works in "multi", returning a browserecord is a bit confusing.
The caller doesn't really know if he gets everything, everything that was set as "just done", or everything that was set as "just done" (and he should not have to read the whole code to find out).

I think a boolean when "everything worked" is a bit better.
But for that you have to check that self matches the steps marked as complete (as again, it works in multi, so just using "bool" on the result could be confusing, as it may not have worked for all steps).

Do you see what I mean?

Reply:
If a step was already done, but you called the action on it and another one, it won't be returned from the progress, so you'd return `False`, which can be confusing, as it was toggled for one of the steps.

One of the return values from the late Enum was "JUST_DONE" which means "at least one got validated" because it could mean that something needs to be synced subsequently.
Returning the exact steps that changed can also support that. However, returning False will not be helpful.

I'll push the other change, then perhaps we can solve this IRL tomorrow if necessary :)